core/server/master/src/main/java/alluxio/master/file/meta/SimpleInodeLockList.java

```diff
@@ -32,20 +31,30 @@
 public class SimpleInodeLockList implements InodeLockList {
   private static final Logger LOG = LoggerFactory.getLogger(SimpleInodeLockList.class);
 
+  private static final int NO_WRITE_LOCK = -1;
   private static final int INITIAL_CAPACITY = 8;
   private static final Edge ROOT_EDGE = new Edge(-1, "");
 
   private final InodeLockManager mInodeLockManager;
 
   /**
-   * Entries for each lock in the lock list. The entries always alternate between InodeEntry and
-   * EdgeEntry.
+   * Inode list.
    */
-  private List<Entry> mEntries;
+  private List<Inode> mInodes;
   /**
-   * The index of the first write lock entry. All locks before this index are read locks, and all
-   * locks after and including this index are write locks. If all locks are read locks,
-   * mFirstWriteLockIndex == mEntries.size().
+   * If last lock in mLocks is an edge lock, then this is the edge.
+   * Otherwise, null.
+   */
+  private Edge mLastEdge;
+  /**
+   * Lock list. The locks always alternate between Inode lock and Edge lock.
```

Comment:
Is inode lock always first?

Reply:
Added comment `The first lock can be either Inode or Edge lock.`